### DISCUSSION TITLE
2D-RVE from EBSD scan

### DISCUSSION THREAD

**Discussion by Anmol-Bakshi** [neperfepx/neper#898] [Q&A] [answered]
I am using the following command to generate 2D-RVE from my EBSD scan:
```
neper -T -n from_morpho -morpho "size:file(size.txt),centroid:file(coordinates.txt)" -dim 2 -ori "file(orientations.txt,des=euler-bunge)" -oridescriptor "euler-bunge:active" -domain "square(8.11,1.73)" -o simulation4
```
where, orientations, coordinates and size are as:
```
Grain Number	 PHI-1	PHI	PHI-2	x (in mm)  	y (in mm)	    Grain Diameter (in mm)
1		        96.8	      43.6	 4.3	         0.065		1.595		0.335
2		      171.9	     17.0	33.1	         0.130		0.290		0.515
3		      276.9	     34.7	18.4	         1.750		0.955		2.664
4		     108.1	     35.8	40.8	         5.879		0.980		3.149
5		     318.8	     37.2	39.8	         8.029		0.085		0.292
```

Overall, I am getting a good match with my EBSD scan image, but in my EBSD scan image Grain-1 and Grain-2 are not touching each other, whereas in NEPER generated image they are touching each other. How to improve it for a more prefect match.

![NEPER image](https://github.com/user-attachments/assets/dedf8aff-6b7c-4c81-a2c4-a403f1d4fd4d)
![Merged-Image-EBSD_with-grain-markings](https://github.com/user-attachments/assets/c4b5ef27-1751-45e2-93e2-ed6e010f36c7)

**Comment by rquey** [comment, marked as answer]
Since you have an EBSD map, the best would be to use it directly instead of considering only the centroids and volumes. To do so, write the EBSD map as a [.tesr file](https://neper.info/doc/fileformat.html#raster-tessellation-file-tesr) and use this file to generate a `.tess`.

You may be interested in this tutorial: [Processing EBSD-type data](https://neper.info/doc/tutorials/ebsd_process.html), and maybe this one too: [Meshing a Polycrystal Image Obtained by Synchrotron X-Ray Diffraction](https://neper.info/doc/tutorials/morpho_tesr_mesh.html#meshing-a-polycrystal-image-obtained-by-synchrotron-x-ray-diffraction) (which is for meshing and 3D data, so, more complicated than what you'll have to do for 2D data).

If you are interested in meshing, also note that 2D maps can also be meshed directly (to account for the grain boundary curvatures). So, you could do something like this:
```
neper -M my_file.tesr
```
